### PR TITLE
Add Task data to BigQuery

### PR DESCRIPTION
**Pull request by keyonghan**
This PR inserts Task data to BigQuery at the same time when updating Task data to Datastore. 

Inserting Task data to BigQuery happens when tasks' statuses are final: either Succeeded or Failed twice (config.maxTaskRetries). This way, one task will only be inserted once to BigQuery.

### CODE REVIEW

**Comment by keyonghan** [comment]
Updated as suggested in #512

**Comment by keyonghan** [comment]
blocked by #512 : fake_tabledata_resouce.dart will be merged in #512.

**Comment by keyonghan** [comment]
@digiter please help take a quick review for this one. I have made changes following those in #512 . Thank you!

File: app_dart/test/request_handlers/update_task_status_test.dart [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      final Commit commit = Commit(key: config.db.emptyKey.append(Commit, id: 'flutter/flutter/7d03371610c07953a5def50d500045941de516b8'));
+      final Task task = Task(key: commit.key.append(Task, id: 4590522719010816), commitKey: commit.key);
+      config.db.values[commit.key] = commit;
+      config.db.values[task.key] = task;
+
+      final TimeSeries timeSeries =
+          TimeSeries(key: config.db.emptyKey.append(TimeSeries, id: 'cubic_bezier_perf__timeline_summary.90th_percentile_frame_build_time_millis'));
+      config.db.values[timeSeries.key] = timeSeries;
+
+      expect(task.status, null);
```
Comment (digiter) [inline app_dart/test/request_handlers/update_task_status_test.dart]: expect(..., isNull)

File: app_dart/lib/src/request_handlers/update_task_status.dart [hunk 2]
```diff
@@ -75,16 +89,23 @@ class UpdateTaskStatus extends ApiRequestHandler<UpdateTaskStatusResponse> {
...
 
-    await config.db.withTransaction<void>((Transaction transaction) async {
+    await datastore.db.withTransaction<void>((Transaction transaction) async {
       transaction.queueMutations(inserts: <Task>[task]);
       await transaction.commit();
     });
 
+    /// Insert data to [BigQuery] when task status is fianlized
+    /// 
+    /// [endTimestamp] greater than 0 is a good final-status flag
```
Comment (digiter) [inline app_dart/lib/src/request_handlers/update_task_status.dart]: Nit: this comment is simply repeating the code, could be removed.

File: app_dart/test/request_handlers/update_task_status_test.dart (hunk 1 above)
Comment (keyonghan) [reply app_dart/test/request_handlers/update_task_status_test.dart]: Updated.

File: app_dart/lib/src/request_handlers/update_task_status.dart (hunk 2 above)
Comment (keyonghan) [reply app_dart/lib/src/request_handlers/update_task_status.dart]: removed.

File: app_dart/lib/src/request_handlers/update_task_status.dart [hunk 3]
```diff
@@ -18,23 +19,34 @@ import '../request_handling/api_request_handler.dart';
...
+    }) : super(config: config, authenticationProvider: authenticationProvider);
+
+  final DatastoreServiceProvider datastoreProvider;
 
   static const String taskKeyParam = 'TaskKey';
   static const String newStatusParam = 'NewStatus';
   static const String resultsParam = 'ResultData';
   static const String scoreKeysParam = 'BenchmarkScoreKeys';
+  
+  /// const variables for [BigQuery] operations
```
Comment (digiter) [inline app_dart/lib/src/request_handlers/update_task_status.dart]: Nit: Although this isn't flutter/flutter repo, but it's nice to use the correct grammar. https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use-correct-grammar

File: app_dart/lib/src/request_handlers/update_task_status.dart [hunk 4]
```diff
@@ -105,6 +123,39 @@ class UpdateTaskStatus extends ApiRequestHandler<UpdateTaskStatusResponse> {
...
+        'Attempts': task.attempts,
+        'IsFlaky': task.isFlaky,
+        'TimeoutInMinutes': task.timeoutInMinutes,
+        'RequiredCapabilities': task.requiredCapabilities?.join(','),
+        'StageName': task.stageName,
+        'Status': task.status,
+      },
+    });
+
+    /// [rows] to be inserted to [BigQuery]
```
Comment (digiter) [inline app_dart/lib/src/request_handlers/update_task_status.dart]: Nit: Also applied here.

**Pull request merged**